torch/csrc/onnx/onnx.h

```diff
@@ -186,6 +192,58 @@ class TensorProto : public MicroProto<onnx_TensorProto> {
   void set_data_type(onnx_TensorProto_DataType t) { proto.has_data_type = true; proto.data_type = t; }
 };
 
+class TypeProtoTensorShapeProto : public MicroProto<onnx_TypeProto_TensorShapeProto> {
```

Comment:
Sigh, nested structs. Maybe we should push back and demand we never have nested structs in the proto.

Reply:
Right now... the proto becomes more complicated...